src/benchmark.zig

```diff
@@ -234,48 +245,65 @@ const TimedQueue = struct {
         const now = std.time.milliTimestamp();
-        assert(results.len == 0);
-        if (log_level == .debug) {
-            const response = std.mem.bytesAsSlice(CreateAccountsResult, results);
-            log.debug("{o}", .{response});
-        }
+        if (results) |value| {
+            if (log_level == .debug) {
+                const response = std.mem.bytesAsSlice(CreateAccountsResult, value);
+                log.debug("response={o}", .{response});
+            }
 
-        const self: *TimedQueue = @intToPtr(*TimedQueue, @intCast(usize, user_data));
-        const completed_batch: ?Batch = self.batches.pop();
-        assert(completed_batch != null);
-        assert(completed_batch.?.operation == operation);
-
-        log.debug("completed batch operation={} start={}", .{ completed_batch.?.operation, self.batch_start });
-        const latency = now - self.batch_start.?;
-        switch (operation) {
-            .create_accounts => {},
-            .create_transfers => {
-                if (latency > self.max_transfers_latency) {
-                    self.max_transfers_latency = latency;
-                }
-            },
-            .commit_transfers => {
-                if (latency > self.max_commits_latency) {
-                    self.max_commits_latency = latency;
-                }
-            },
-            else => unreachable,
-        }
+            const self: *TimedQueue = @intToPtr(*TimedQueue, @intCast(usize, user_data));
+            const completed_batch: ?Batch = self.batches.pop();
+            assert(completed_batch != null);
+            assert(completed_batch.?.operation == operation);
 
-        var batch: ?Batch = self.batches.peek();
-        if (batch) |*next_batch| {
-            self.batch_start = std.time.milliTimestamp();
-            self.client.request(
-                @intCast(u128, @ptrToInt(self)),
-                TimedQueue.lap,
-                next_batch.operation,
-                next_batch.data,
-            );
-        } else {
-            log.debug("stopping timer...", .{});
-            self.end = now;
+            log.debug("completed batch operation={} start={}", .{
+                completed_batch.?.operation,
+                self.batch_start,
+            });
+            const latency = now - self.batch_start.?;
+            switch (operation) {
+                .create_accounts => {},
+                .create_transfers => {
+                    if (latency > self.max_transfers_latency) {
+                        self.max_transfers_latency = latency;
+                    }
+                },
+                .commit_transfers => {
+                    if (latency > self.max_commits_latency) {
+                        self.max_commits_latency = latency;
+                    }
+                },
+                else => unreachable,
+            }
+
+            var batch: ?Batch = self.batches.peek();
+            var message = self.client.get_message() orelse {
```

Comment:
We could move this into the `next_batch` branch since that is where we need it. Otherwise we might panic if there's no free message even though we've completed the benchmark.